landoui/revisions.py

```diff
@@ -227,10 +227,31 @@ def revisions_handler(revision_id, diff_id=None):
 
 @revisions.route('/request-sec-approval', methods=('POST', ))
 def sec_approval_request_handler():
+    if not current_app.config.get("ENABLE_SEC_APPROVAL"):
+        abort(404)
+
+    token = get_phabricator_api_token()
+    if not token:
+        # The user has not set their API Token. Lando API will return an
+        # error if it is missing.
+        errors = make_form_error(
+            'You must set your Phabricator API token in the Lando User '
+            'Settings to request sec-approval'
+        )
+        return jsonify(errors=errors), 400
+
+    api = LandoAPI(
+        current_app.config['LANDO_API_URL'],
+        auth0_access_token=session.get('access_token'),
+        phabricator_api_token=token,
+    )
+
     form = SecApprovalRequestForm()
-    # Errors are stored in a dict containing a list of errors for each field.
+
     if not is_user_authenticated():
```

Comment:
Probably best to move this auth check up next to the Phabricator token check. Keeping all the login/permission stuff early and together is less fragile IMO.

Reply:
Done, looks nicer thanks.